src/sql/workbench/services/diagnostics/browser/errorDiagnosticsService.ts

```diff
@@ -18,11 +19,16 @@ export class ErrorDiagnosticsService implements IErrorDiagnosticsService {
 		@ILogService private readonly _logService: ILogService
 	) { }
 
-	public async tryHandleConnectionError(errorCode: number, errorMessage: string, providerId: string, connection: interfaces.IConnectionProfile): Promise<azdata.diagnostics.ConnectionDiagnosticsResult> {
+	public async tryHandleConnectionError(connectionResult: IConnectionResult, providerId: string, connection: interfaces.IConnectionProfile): Promise<azdata.diagnostics.ConnectionDiagnosticsResult> {
 		let result = { handled: false };
 		let provider = this._providers[providerId]
 		if (provider) {
-			result = await provider.handleConnectionError(errorCode, errorMessage, Utils.convertToRpcConnectionProfile(connection));
+			let errorInfo: azdata.diagnostics.IErrorInformation = {
+				errorCode: connectionResult.errorCode,
```

Comment:
Will these always be defined? Make sure the typings are correct here - or even consider defaulting to a value like an empty string at least if they aren't.

Reply:
They all need to be set even if undefined.. I'll add a fallback to empty string.